lib/awful/widget/common.lua.in

```diff
@@ -74,10 +74,14 @@ function common.list_update(w, buttons, label, data, objects)
             }
         end
 
-        local text, bg, bg_image, icon = label(o)
+        local text, bg, bg_image, icon, icon_only = label(o)
```

Comment:
I find it problematic that the API / return values of `label` gets changed.
(This should be changed in the documentation probably then, and maybe we could use a table instead.)

But for this case the `icon_only` can be derived from `text ~= nil and text ~= ""`, and does not need to be returned from `label`.